saml/src/main/java/com/linecorp/armeria/server/saml/KeyStoreCredentialResolverBuilder.java

```diff
@@ -18,31 +18,68 @@
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
+import java.io.File;
+import java.io.FileInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.security.GeneralSecurityException;
 import java.security.KeyStore;
 import java.util.HashMap;
 import java.util.Map;
 
+import javax.annotation.Nullable;
+
 import org.opensaml.security.credential.CredentialResolver;
 import org.opensaml.security.credential.impl.KeyStoreCredentialResolver;
 
 /**
  * A builder class which creates a new {@link KeyStoreCredentialResolver} instance.
  */
 public final class KeyStoreCredentialResolverBuilder {
+    @Nullable
+    private final File file;
+    @Nullable
+    private final String resourcePath;
+    @Nullable
+    private final ClassLoader classLoader;
 
-    private final String path;
-    private final String keystorePassword;
+    private String type = KeyStore.getDefaultType();
+    @Nullable
+    private String password;
     private final Map<String, String> keyPasswords = new HashMap<>();
 
     /**
-     * Creates a builder instance for {@link KeyStoreCredentialResolver}.
+     * Creates a builder with the specified {@link File}.
+     */
+    public KeyStoreCredentialResolverBuilder(File file) {
+        this.file = requireNonNull(file, "file");
+        resourcePath = null;
+        classLoader = null;
+    }
+
+    /**
+     * Creates a builder with the specified {@code resourcePath} and {@link ClassLoader}.
+     */
+    public KeyStoreCredentialResolverBuilder(String resourcePath, ClassLoader classLoader) {
+        this.resourcePath = requireNonNull(resourcePath, "path");
+        this.classLoader = requireNonNull(classLoader, "classLoader");
+        file = null;
+    }
+
+    /**
+     * Sets a type of the {@link KeyStore}.
```

Comment:
How about adding `If not set, the default value retrieved from {@link KeyStore#getDefaultType()} will be used.`